gridcomps/ExtData3G/PrimaryExport.F90

```diff
@@ -49,6 +55,9 @@ function new_PrimaryExport(export_var, file_var, file_selector) result(primary_e
       call right_node%set_node_side(NODE_RIGHT)
       call primary_export%bracket%set_node(NODE_LEFT, left_node)
       call primary_export%bracket%set_node(NODE_RIGHT, right_node)
+      call file_selector%get_file_template(file_template)
+      primary_export%client_collection_id = i_clients%add_data_collection(file_template, _RC)
```

Comment:
Are there issues if `file_template` is not allocated?  I see that in the getter in the line above that is a possibility, so wanted to double check.

Reply:
ok - but then the implementation of `get_file_template()` is a bit weird in that it checks for allocatable but does not fail if not allocated.      Feels only half guarded at that point.  But minor issue for now.